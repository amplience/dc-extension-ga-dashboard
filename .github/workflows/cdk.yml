name: CDK

on: [push, pull_request]

defaults:
  run:
    working-directory: cdk

jobs:
  build:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [12.x]

    steps:
      - uses: actions/checkout@v1

      - name: Get npm cache directory
        id: npm-cache
        run: |
          echo "::set-output name=dir::$(npm config get cache)"
      - uses: actions/cache@v2
        with:
          path: ${{ steps.npm-cache.outputs.dir }}
          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}

      - name: Use Node.js ${{ matrix.node-version }}
        uses: actions/setup-node@v1
        with:
          node-version: ${{ matrix.node-version }}

      - name: npm install, build, and test
        run: |
          echo "//npm.pkg.github.com/:_authToken=${{ secrets.GH_READ_PACKAGE_PAT }}" > ~/.npmrc
          npm ci
          npm run build --if-present
          npm test
        env:
          CI: true
