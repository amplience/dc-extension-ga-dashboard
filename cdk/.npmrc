registry=https://npm.pkg.github.com/amplience