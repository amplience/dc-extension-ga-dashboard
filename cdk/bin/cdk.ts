#!/usr/bin/env node
import 'source-map-support/register';
import * as cdk from '@aws-cdk/core';
import { AmplifyStack } from '../lib/amplify-stack';
import { SupportedSourceCodeProviders } from '@amplience/amp-cdk-amplify';

const app = new cdk.App();

export const devStack = new AmplifyStack(app, 'dev-dc-extension-ga-dashboard', {
  owner: 'amplience',
  repository: 'dc-extension-ga-dashboard',
  oauthTokenSecretId: 'qa-tupperware-github-pat',
  branch: 'master',
  sourceCodeProvider: SupportedSourceCodeProviders.GITHUB,
  domain: 'dev-dc-extension-ga-dashboard.dev.adis.ws',
  buildPullRequestsPreview: true,
  environmentVariables: {},
  tagging: {
    application: 'dc-extension-ga-dashboard',
    applicationFunction: 'frontend',
    confidentiality: 'public',
    env: 'dev',
    owner: 'tupperware',
    product: 'dc',
    zone: 'public',
  },
});

export const qaStack = new AmplifyStack(app, 'qa-dc-extension-ga-dashboard', {
  owner: 'amplience',
  repository: 'dc-extension-ga-dashboard',
  oauthTokenSecretId: 'qa-tupperware-github-pat',
  branch: 'master',
  sourceCodeProvider: SupportedSourceCodeProviders.GITHUB,
  domain: 'qa-dc-extension-ga-dashboard.dev.adis.ws',
  buildPullRequestsPreview: true,
  environmentVariables: {},
  tagging: {
    application: 'dc-extension-ga-dashboard',
    applicationFunction: 'frontend',
    confidentiality: 'public',
    env: 'qa',
    owner: 'tupperware',
    product: 'dc',
    zone: 'public',
  },
});

export const prodStack = new AmplifyStack(
  app,
  'prod-dc-extension-ga-dashboard',
  {
    owner: 'amplience',
    repository: 'dc-extension-ga-dashboard',
    oauthTokenSecretId: 'qa-tupperware-github-pat',
    branch: 'production',
    sourceCodeProvider: SupportedSourceCodeProviders.GITHUB,
    domain: 'dc-extension-ga-dashboard.amplience.net',
    buildPullRequestsPreview: true,
    environmentVariables: {},
    tagging: {
      application: 'dc-extension-ga-dashboard',
      applicationFunction: 'frontend',
      confidentiality: 'public',
      env: 'prod',
      owner: 'tupperware',
      product: 'dc',
      zone: 'public',
    },
  }
);
