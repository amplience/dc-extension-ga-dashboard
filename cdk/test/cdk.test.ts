import { SynthUtils } from '@aws-cdk/assert';
import { qaStack, devStack, prodStack } from '../bin/cdk';

describe('snapshot tests', () => {
  describe('qaStack', () => {
    it('should match dev snapshot', () => {
      expect(SynthUtils.toCloudFormation(devStack)).toMatchSnapshot();
    });

    it('should match qa snapshot', () => {
      expect(SynthUtils.toCloudFormation(qaStack)).toMatchSnapshot();
    });

    it('should match prod snapshot', () => {
      expect(SynthUtils.toCloudFormation(prodStack)).toMatchSnapshot();
    });
  });
});
