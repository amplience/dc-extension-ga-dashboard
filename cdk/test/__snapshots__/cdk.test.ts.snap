// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests qaStack should match dev snapshot 1`] = `
Object {
  "Outputs": Object {
    "devdcextensiongadashboardampamplifyAmplifyAppIdF2F99189": Object {
      "Export": Object {
        "Name": "dev-dc-extension-ga-dashboard-amp-amplify-AmplifyAppId",
      },
      "Value": Object {
        "Fn::GetAtt": Array [
          "devdcextensiongadashboardampamplifyD20367BF",
          "AppId",
        ],
      },
    },
    "devdcextensiongadashboardampamplifyCertificateRecordDD187031": Object {
      "Export": Object {
        "Name": "dev-dc-extension-ga-dashboard-amp-amplify-CertificateRecord",
      },
      "Value": Object {
        "Fn::GetAtt": Array [
          "devdcextensiongadashboardampamplifydevdcextensiongadashboarddevadisws3B62BBBF",
          "CertificateRecord",
        ],
      },
    },
    "devdcextensiongadashboardampamplifyDomainName5A94C755": Object {
      "Export": Object {
        "Name": "dev-dc-extension-ga-dashboard-amp-amplify-DomainName",
      },
      "Value": Object {
        "Fn::GetAtt": Array [
          "devdcextensiongadashboardampamplifydevdcextensiongadashboarddevadisws3B62BBBF",
          "DomainName",
        ],
      },
    },
  },
  "Resources": Object {
    "devdcextensiongadashboardampamplifyD20367BF": Object {
      "Properties": Object {
        "IAMServiceRole": Object {
          "Fn::GetAtt": Array [
            "devdcextensiongadashboardampamplifyRole01830EFA",
            "Arn",
          ],
        },
        "Name": "dev-dc-extension-ga-dashboard-amp-amplify",
        "OauthToken": "{{resolve:secretsmanager:qa-tupperware-github-pat:SecretString:::}}",
        "Repository": "https://github.com/amplience/dc-extension-ga-dashboard",
        "Tags": Array [
          Object {
            "Key": "amp:application",
            "Value": "dc-extension-ga-dashboard",
          },
          Object {
            "Key": "amp:application-function",
            "Value": "frontend",
          },
          Object {
            "Key": "amp:confidentiality",
            "Value": "public",
          },
          Object {
            "Key": "amp:owner",
            "Value": "tupperware",
          },
          Object {
            "Key": "amp:product",
            "Value": "dc",
          },
          Object {
            "Key": "amp:zone",
            "Value": "public",
          },
          Object {
            "Key": "env",
            "Value": "dev",
          },
        ],
      },
      "Type": "AWS::Amplify::App",
    },
    "devdcextensiongadashboardampamplifyRole01830EFA": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "amplify.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": Array [
          Object {
            "Key": "amp:application",
            "Value": "dc-extension-ga-dashboard",
          },
          Object {
            "Key": "amp:application-function",
            "Value": "frontend",
          },
          Object {
            "Key": "amp:confidentiality",
            "Value": "public",
          },
          Object {
            "Key": "amp:owner",
            "Value": "tupperware",
          },
          Object {
            "Key": "amp:product",
            "Value": "dc",
          },
          Object {
            "Key": "amp:zone",
            "Value": "public",
          },
          Object {
            "Key": "env",
            "Value": "dev",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "devdcextensiongadashboardampamplifydevdcextensiongadashboarddevadisws3B62BBBF": Object {
      "Properties": Object {
        "AppId": Object {
          "Fn::GetAtt": Array [
            "devdcextensiongadashboardampamplifyD20367BF",
            "AppId",
          ],
        },
        "DomainName": "dev-dc-extension-ga-dashboard.dev.adis.ws",
        "SubDomainSettings": Array [
          Object {
            "BranchName": Object {
              "Fn::GetAtt": Array [
                "devdcextensiongadashboardampamplifymaster669128F6",
                "BranchName",
              ],
            },
            "Prefix": "",
          },
        ],
      },
      "Type": "AWS::Amplify::Domain",
    },
    "devdcextensiongadashboardampamplifymaster669128F6": Object {
      "Properties": Object {
        "AppId": Object {
          "Fn::GetAtt": Array [
            "devdcextensiongadashboardampamplifyD20367BF",
            "AppId",
          ],
        },
        "BranchName": "master",
        "EnableAutoBuild": true,
        "EnablePullRequestPreview": true,
        "Tags": Array [
          Object {
            "Key": "amp:application",
            "Value": "dc-extension-ga-dashboard",
          },
          Object {
            "Key": "amp:application-function",
            "Value": "frontend",
          },
          Object {
            "Key": "amp:confidentiality",
            "Value": "public",
          },
          Object {
            "Key": "amp:owner",
            "Value": "tupperware",
          },
          Object {
            "Key": "amp:product",
            "Value": "dc",
          },
          Object {
            "Key": "amp:zone",
            "Value": "public",
          },
          Object {
            "Key": "env",
            "Value": "dev",
          },
        ],
      },
      "Type": "AWS::Amplify::Branch",
    },
  },
}
`;

exports[`snapshot tests qaStack should match prod snapshot 1`] = `
Object {
  "Outputs": Object {
    "proddcextensiongadashboardampamplifyAmplifyAppIdF55EC2E0": Object {
      "Export": Object {
        "Name": "prod-dc-extension-ga-dashboard-amp-amplify-AmplifyAppId",
      },
      "Value": Object {
        "Fn::GetAtt": Array [
          "proddcextensiongadashboardampamplify5AFE810A",
          "AppId",
        ],
      },
    },
    "proddcextensiongadashboardampamplifyCertificateRecordE757EAAA": Object {
      "Export": Object {
        "Name": "prod-dc-extension-ga-dashboard-amp-amplify-CertificateRecord",
      },
      "Value": Object {
        "Fn::GetAtt": Array [
          "proddcextensiongadashboardampamplifydcextensiongadashboardampliencenet177C1CD6",
          "CertificateRecord",
        ],
      },
    },
    "proddcextensiongadashboardampamplifyDomainName47777F9B": Object {
      "Export": Object {
        "Name": "prod-dc-extension-ga-dashboard-amp-amplify-DomainName",
      },
      "Value": Object {
        "Fn::GetAtt": Array [
          "proddcextensiongadashboardampamplifydcextensiongadashboardampliencenet177C1CD6",
          "DomainName",
        ],
      },
    },
  },
  "Resources": Object {
    "proddcextensiongadashboardampamplify5AFE810A": Object {
      "Properties": Object {
        "IAMServiceRole": Object {
          "Fn::GetAtt": Array [
            "proddcextensiongadashboardampamplifyRoleE0693ECE",
            "Arn",
          ],
        },
        "Name": "prod-dc-extension-ga-dashboard-amp-amplify",
        "OauthToken": "{{resolve:secretsmanager:qa-tupperware-github-pat:SecretString:::}}",
        "Repository": "https://github.com/amplience/dc-extension-ga-dashboard",
        "Tags": Array [
          Object {
            "Key": "amp:application",
            "Value": "dc-extension-ga-dashboard",
          },
          Object {
            "Key": "amp:application-function",
            "Value": "frontend",
          },
          Object {
            "Key": "amp:confidentiality",
            "Value": "public",
          },
          Object {
            "Key": "amp:owner",
            "Value": "tupperware",
          },
          Object {
            "Key": "amp:product",
            "Value": "dc",
          },
          Object {
            "Key": "amp:zone",
            "Value": "public",
          },
          Object {
            "Key": "env",
            "Value": "prod",
          },
        ],
      },
      "Type": "AWS::Amplify::App",
    },
    "proddcextensiongadashboardampamplifyRoleE0693ECE": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "amplify.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": Array [
          Object {
            "Key": "amp:application",
            "Value": "dc-extension-ga-dashboard",
          },
          Object {
            "Key": "amp:application-function",
            "Value": "frontend",
          },
          Object {
            "Key": "amp:confidentiality",
            "Value": "public",
          },
          Object {
            "Key": "amp:owner",
            "Value": "tupperware",
          },
          Object {
            "Key": "amp:product",
            "Value": "dc",
          },
          Object {
            "Key": "amp:zone",
            "Value": "public",
          },
          Object {
            "Key": "env",
            "Value": "prod",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "proddcextensiongadashboardampamplifydcextensiongadashboardampliencenet177C1CD6": Object {
      "Properties": Object {
        "AppId": Object {
          "Fn::GetAtt": Array [
            "proddcextensiongadashboardampamplify5AFE810A",
            "AppId",
          ],
        },
        "DomainName": "dc-extension-ga-dashboard.amplience.net",
        "SubDomainSettings": Array [
          Object {
            "BranchName": Object {
              "Fn::GetAtt": Array [
                "proddcextensiongadashboardampamplifyproduction11819A19",
                "BranchName",
              ],
            },
            "Prefix": "",
          },
        ],
      },
      "Type": "AWS::Amplify::Domain",
    },
    "proddcextensiongadashboardampamplifyproduction11819A19": Object {
      "Properties": Object {
        "AppId": Object {
          "Fn::GetAtt": Array [
            "proddcextensiongadashboardampamplify5AFE810A",
            "AppId",
          ],
        },
        "BranchName": "production",
        "EnableAutoBuild": true,
        "EnablePullRequestPreview": true,
        "Tags": Array [
          Object {
            "Key": "amp:application",
            "Value": "dc-extension-ga-dashboard",
          },
          Object {
            "Key": "amp:application-function",
            "Value": "frontend",
          },
          Object {
            "Key": "amp:confidentiality",
            "Value": "public",
          },
          Object {
            "Key": "amp:owner",
            "Value": "tupperware",
          },
          Object {
            "Key": "amp:product",
            "Value": "dc",
          },
          Object {
            "Key": "amp:zone",
            "Value": "public",
          },
          Object {
            "Key": "env",
            "Value": "prod",
          },
        ],
      },
      "Type": "AWS::Amplify::Branch",
    },
  },
}
`;

exports[`snapshot tests qaStack should match qa snapshot 1`] = `
Object {
  "Outputs": Object {
    "qadcextensiongadashboardampamplifyAmplifyAppIdF183F3A3": Object {
      "Export": Object {
        "Name": "qa-dc-extension-ga-dashboard-amp-amplify-AmplifyAppId",
      },
      "Value": Object {
        "Fn::GetAtt": Array [
          "qadcextensiongadashboardampamplifyA5EF52A7",
          "AppId",
        ],
      },
    },
    "qadcextensiongadashboardampamplifyCertificateRecord06FCA6D5": Object {
      "Export": Object {
        "Name": "qa-dc-extension-ga-dashboard-amp-amplify-CertificateRecord",
      },
      "Value": Object {
        "Fn::GetAtt": Array [
          "qadcextensiongadashboardampamplifyqadcextensiongadashboarddevadiswsC9D483E6",
          "CertificateRecord",
        ],
      },
    },
    "qadcextensiongadashboardampamplifyDomainNameADED8974": Object {
      "Export": Object {
        "Name": "qa-dc-extension-ga-dashboard-amp-amplify-DomainName",
      },
      "Value": Object {
        "Fn::GetAtt": Array [
          "qadcextensiongadashboardampamplifyqadcextensiongadashboarddevadiswsC9D483E6",
          "DomainName",
        ],
      },
    },
  },
  "Resources": Object {
    "qadcextensiongadashboardampamplifyA5EF52A7": Object {
      "Properties": Object {
        "IAMServiceRole": Object {
          "Fn::GetAtt": Array [
            "qadcextensiongadashboardampamplifyRoleCD97F8FB",
            "Arn",
          ],
        },
        "Name": "qa-dc-extension-ga-dashboard-amp-amplify",
        "OauthToken": "{{resolve:secretsmanager:qa-tupperware-github-pat:SecretString:::}}",
        "Repository": "https://github.com/amplience/dc-extension-ga-dashboard",
        "Tags": Array [
          Object {
            "Key": "amp:application",
            "Value": "dc-extension-ga-dashboard",
          },
          Object {
            "Key": "amp:application-function",
            "Value": "frontend",
          },
          Object {
            "Key": "amp:confidentiality",
            "Value": "public",
          },
          Object {
            "Key": "amp:owner",
            "Value": "tupperware",
          },
          Object {
            "Key": "amp:product",
            "Value": "dc",
          },
          Object {
            "Key": "amp:zone",
            "Value": "public",
          },
          Object {
            "Key": "env",
            "Value": "qa",
          },
        ],
      },
      "Type": "AWS::Amplify::App",
    },
    "qadcextensiongadashboardampamplifyRoleCD97F8FB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "amplify.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Tags": Array [
          Object {
            "Key": "amp:application",
            "Value": "dc-extension-ga-dashboard",
          },
          Object {
            "Key": "amp:application-function",
            "Value": "frontend",
          },
          Object {
            "Key": "amp:confidentiality",
            "Value": "public",
          },
          Object {
            "Key": "amp:owner",
            "Value": "tupperware",
          },
          Object {
            "Key": "amp:product",
            "Value": "dc",
          },
          Object {
            "Key": "amp:zone",
            "Value": "public",
          },
          Object {
            "Key": "env",
            "Value": "qa",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "qadcextensiongadashboardampamplifymaster464DA772": Object {
      "Properties": Object {
        "AppId": Object {
          "Fn::GetAtt": Array [
            "qadcextensiongadashboardampamplifyA5EF52A7",
            "AppId",
          ],
        },
        "BranchName": "master",
        "EnableAutoBuild": true,
        "EnablePullRequestPreview": true,
        "Tags": Array [
          Object {
            "Key": "amp:application",
            "Value": "dc-extension-ga-dashboard",
          },
          Object {
            "Key": "amp:application-function",
            "Value": "frontend",
          },
          Object {
            "Key": "amp:confidentiality",
            "Value": "public",
          },
          Object {
            "Key": "amp:owner",
            "Value": "tupperware",
          },
          Object {
            "Key": "amp:product",
            "Value": "dc",
          },
          Object {
            "Key": "amp:zone",
            "Value": "public",
          },
          Object {
            "Key": "env",
            "Value": "qa",
          },
        ],
      },
      "Type": "AWS::Amplify::Branch",
    },
    "qadcextensiongadashboardampamplifyqadcextensiongadashboarddevadiswsC9D483E6": Object {
      "Properties": Object {
        "AppId": Object {
          "Fn::GetAtt": Array [
            "qadcextensiongadashboardampamplifyA5EF52A7",
            "AppId",
          ],
        },
        "DomainName": "qa-dc-extension-ga-dashboard.dev.adis.ws",
        "SubDomainSettings": Array [
          Object {
            "BranchName": Object {
              "Fn::GetAtt": Array [
                "qadcextensiongadashboardampamplifymaster464DA772",
                "BranchName",
              ],
            },
            "Prefix": "",
          },
        ],
      },
      "Type": "AWS::Amplify::Domain",
    },
  },
}
`;
